client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/OutboundLineEdit.tsx

```diff
@@ -1,197 +1,130 @@
+import { SelectItem } from './SelectItem';
+import { Allocation } from './Allocation';
+import { useOpenedWithBarcode } from './hooks/useOpenedWithBarcode';
+import { useAllocationContext } from './allocation/useAllocationContext';
+import { useSaveOutboundLines } from '../../api/hooks/useSaveOutboundLines';
+import { getAllocatedUnits } from './allocation/utils';
+
+export type OutboundOpenedWith = { itemId: string } | ScannedBarcode | null;
+
+interface OutboundLineEditProps {
   isOpen: boolean;
   onClose: () => void;
-  draft: Draft | null;
+  openedWith: OutboundOpenedWith;
   mode: ModalMode | null;
+  status: InvoiceNodeStatus;
+  invoiceId: string;
 }
 
-const useFocusNumberOfPacksInput = (draft: Draft | null) => {
-  const batch = draft?.barcode?.batch;
-
-  useEffect(() => {
-    setTimeout(() => {
-      if (!batch) return;
-      const input = document.getElementById(getPackQuantityCellId(batch));
-      if (input) {
-        input.focus();
-      }
-    }, 500);
-  }, [batch]);
-};
-
-export const OutboundLineEdit: React.FC<ItemDetailsModalProps> = ({
+export const OutboundLineEdit = ({
   isOpen,
-  onClose,
-  draft,
+  onClose: closeModal,
+  openedWith,
   mode,
-}) => {
-  const item = !draft ? null : (draft.item ?? null);
+  status,
+  invoiceId,
+}: OutboundLineEditProps) => {
   const t = useTranslation();
-  const { info } = useNotification();
+  const { info, warning } = useNotification();
+  const [itemId, setItemId] = useState(openedWith?.itemId);
+
+  const onClose = () => {
+    clear();
+    closeModal();
+  };
   const { Modal } = useDialog({ isOpen, onClose, disableBackdrop: true });
-  const [currentItem, setCurrentItem] = useBufferState(item);
-  const [isAutoAllocated, setIsAutoAllocated] = useState(false);
-  const [okDisabled, setOkDisabled] = useState(false);
 
-  const { mutateAsync: insertBarcode } = useOutbound.utils.barcodeInsert();
-  const { status, currency, otherParty } = useOutbound.document.fields([
-    'status',
-    'currency',
-    'otherParty',
-  ]);
-  const { mutateAsync } = useOutbound.line.save(status);
-  const isDisabled = useOutbound.utils.isDisabled();
-  const {
-    draftStockOutLines,
-    updateQuantity,
-    setDraftStockOutLines,
-    isLoading,
-  } = useDraftOutboundLines(currentItem);
-  const packSizeController = usePackSizeController(draftStockOutLines);
-  const { next, disabled: nextDisabled } = useNextItem(currentItem?.id);
-  const { isDirty, setIsDirty } = useDirtyCheck();
-  const { warning } = useNotification();
-  const [showZeroQuantityConfirmation, setShowZeroQuantityConfirmation] =
-    useState(false);
-  useFocusNumberOfPacksInput(draft);
+  const { next, disabled: nextDisabled } = useNextItem(itemId);
 
-  const placeholder = draftStockOutLines?.find(
-    ({ type, numberOfPacks }) =>
-      type === InvoiceLineNodeType.UnallocatedStock && numberOfPacks !== 0
+  const { data: itemData, isFetching } = useOutboundLineEditData(
```

Comment:
custom query - all the bits we need for the modal in one query - rather than composing several queries, with loads of state updates after each resolves. Much more efficient for slower networks/devices too 🎉